plugins/TPCSDAction.cpp

```diff
@@ -275,36 +290,38 @@ namespace sim {
       dEInPadRow = 0.0;
       globalTimeAtPadRingCentre = 0.0;
       pathLengthInPadRow = 0.0;
-      CrossingOfPadRingCentre[0] = 0.0;
-      CrossingOfPadRingCentre[1] = 0.0;
-      CrossingOfPadRingCentre[2] = 0.0;
-      MomentumAtPadRingCentre[0] = 0.0;
-      MomentumAtPadRingCentre[1] = 0.0;
-      MomentumAtPadRingCentre[2] = 0.0;
-      StepAtEntranceToPadRing = 0;
+      for (int i = 0; i < 3; i++) {
+        CrossingOfPadRingCentre[i] = 0.0;
+        MomentumAtPadRingCentre[i] = 0.0;
+      }
+      stepAtEntranceToPadRingInfo.isValid = false;
     }
 
-    void DepositHiPtHit(G4Step const* step) // DJ extracted to separate fn
-    {
+    void DepositHiPtHit(stepInfo mainStepInfo) {
+
+      if (!mainStepInfo.isValid || !stepAtEntranceToPadRingInfo.isValid) {
+        std::cerr << "DepositHiPtHit ERROR, invalid stepInfo " << mainStepInfo.isValid << " "
+                  << stepAtEntranceToPadRingInfo.isValid << std::endl;
```

Comment:
Can we go via some dd4hep logging facility for this? I think we should have rather simple access via the `sensitive`: https://dd4hep.web.cern.ch/dd4hep/reference/classdd4hep_1_1sim_1_1Geant4Action.html#a1abdae5d2d2af327f406697322b0d821

Otherwise, we can always go via `dd4hep::printout`